cloud/satellite/satellite_activation_keys.py

```diff
@@ -0,0 +1,365 @@
+    organization      = module.params.get('organization')
+    key_action  = module.params.get('key_action')
+    key_name  = module.params.get('key_name')
+    validate_certs = module.params.get('validate_certs')
+    key_description = module.params.get('key_description')
+    environment = module.params.get('environment')
+    content_view = module.params.get('content_view')
+    max_content_hosts = module.params.get('max_content_hosts')
+    unlimited_content_hosts = module.params.get('unlimited_content_hosts')
+    release_version = module.params.get('release_version')
+    service_level = module.params.get('service_level')
+    auto_attach = module.params.get('auto_attach')
+
+    changed = False
+
+    orga_id = get_organization_id(module)
+    key = get_key(module,orga_id)
+    if 'environment' in locals():
+        env_id = get_environment_id(module,orga_id)
+    if 'content_view' in locals():
+        cv_id = get_content_view_id(module,orga_id)
+
+    url_action = 'https://' + satellite + '/katello/api/activation_keys/'
+    headers = {'Content-Type':'application/json'}
+
+    if 'create' in key_action:
+        payload = {"name":key_name,"organization_id":orga_id}
+        if 'key_description' in locals():
+            payload["description"] = key_description
+        if 'environment' in locals():
+            payload["environment_id"] = env_id
+        if 'content_view' in locals():
+            payload["content_view_id"] = cv_id
+        if 'max_content_hosts' in locals():
+            payload["max_content_hosts"] = max_content_hosts
+        if 'unlimited_content_hosts' in locals():
+            payload["unlimited_content_hosts"] = unlimited_content_hosts
+        if module.check_mode:
+            if key is None:
+                return True,"Key created (check mode)"
+            return False,"No key created (check mode)"
+        if key is not None:
+            return False,"Key already exists."
+        else:
+            resp = open_url(url_action,method="POST",headers=headers,url_username=module.params.get('user'),url_password=module.params.get('pwd'),force_basic_auth=True,data=json.dumps(payload),validate_certs=validate_certs)
+    elif 'delete' in key_action:
+        if key is not None:
+            if module.check_mode:
+                return True,"Key deleted (check mode)"
+            url_action_delete = url_action + str(key["id"])
+            resp = open_url(url_action_delete,method="DELETE",headers=headers,url_username=module.params.get('user'),url_password=module.params.get('pwd'),force_basic_auth=True,validate_certs=validate_certs)
+        else:
+            return False,"No key deleted"
+    elif 'update' in key_action:
+        if key is None:
+            module.fail_json(msg="Key name %s not found." % key_name)
+        url_action_put = url_action + str(key["id"])
+        payload = {"organization_id":orga_id}
+        if ('key_description' in locals()) and (key_description != key["description"]):
```

Comment:
what does locals return?  Ideally a dict, but if this is a list the complexity on this lookup is O(n)